### PR TITLE
fixes #4798 feat(nimbus): Add reject component for review flow

### LINKED ISSUE #4798: Create new rejection reason component for new Review flow
Create new component to report the reason for a rejection to experiment owner when rejected by reviewer.

Use a feature flag to demo the component in Storybook while leaving it disabled in production rendering until final integration of FE / BE. Documentation on how to do this is TBD with EXP-1063, but that issue has some links to pointers & examples.

Figma mockups: https://www.figma.com/file/5SLa07XqQb0ttB1ay1gami/Nimbus-Console-UI?node-id=217334%3A6

!Capture.PNG|thumbnail!

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-1060)
┆Attachments: <a href="https://jira.mozilla.com/secure/attachment/15530/Capture.PNG">Capture.PNG</a>
┆Issue Number: EXP-1060

### PR DESCRIPTION
**Pull request by LZoog**
fixes #4798

Because:
* We need to display rejected review feedback to the user

This commit:
* Creates a new FeedbackRejected component to display when reject feedback is present
* Sets a flag to only display in Storybook at the time of creation

### CODE REVIEW

**Comment by lmorchard** [comment]
FWIW, this is going to conflict with PR #4861, because I did a bunch of refactoring in PageRequestReview to handle the growing number of states for draft status actions

**Comment by LZoog** [comment]
@lmorchard Hmm yeah, I'm cool with either; if this LGTY maybe we can go ahead and get it in, but I wouldn't mind integrating it into your work over there either. I'll take a look this morning at that PR.

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/FeedbackRejected/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { render, screen } from "@testing-library/react";
+import React from "react";
+import FeedbackRejected from ".";
+
+describe("FeedbackRejected", () => {
```
Comment (LZoog) [inline app/experimenter/nimbus-ui/src/components/PageRequestReview/FeedbackRejected/index.test.tsx:9]: Also, I know I could have just stuck an istabul ignore in here, but this seemed simple enough to test for now and will be easy to replace with the "real" (mock) data later.

**Comment by LZoog** [comment]
@lmorchard After checking out your PR, I think I'll wait on yours to get merged and then stick this on top of your changes.

**Comment by LZoog** [comment]
Alright, opened this for review again after refactoring on top of @lmorchard ' recent changes, though there is one state I didn't see in Figma that I'd like to run by Ana.

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/DraftStatusOperations.tsx [hunk 2]
```diff
@@ -118,6 +134,40 @@ export const DraftStatusOperations = ({
...
+            </p>
+            <p className="bg-white rounded border p-3">
+              {rejectFeedback!.rejectReason}
+            </p>
+          </div>
+        </Alert>
+      );
+
+    /* istanbul ignore next until EXP-1055 & EXP-1062 done */
+    case DraftStatusOperationsState.LaunchRejectedByAnotherUser:
```
Comment (LZoog) [inline app/experimenter/nimbus-ui/src/components/PageRequestReview/DraftStatusOperations.tsx:156]: I need to run this one by Ana. I'm not sure what we want to display if the experiment has been rejected but the current user 1) does not have reviewer role and does not own the experiment, and 2) if user has a reviewer role but didn't do the rejecting.

**Review by lmorchard — APPROVED** [review]
LGTM, I'm okay with merging this and further iterating based on how we want to handle rejection reason visibility vs role & ownership

**Pull request merged**